json-encode.go

```diff
@@ -135,50 +135,44 @@ func (cdc *Codec) encodeReflectJSONArrayOrSlice(w io.Writer, info *TypeInfo, rv
 	return err
 }
 
-func safeElem(v reflect.Value) reflect.Value {
-	// As per https://golang.org/pkg/reflect/#Value.Elem
-	// Elem can only be invoked on an interface or a pointer.
-	if v.Kind() == reflect.Interface || v.Kind() == reflect.Ptr {
-		return v.Elem()
-	}
-	return v
-}
-
-func (cdc *Codec) encodeReflectJSONInterface(w io.Writer, info *TypeInfo, rv reflect.Value, opts FieldOptions) error {
+func (cdc *Codec) encodeReflectJSONInterface(w io.Writer, iinfo *TypeInfo, rv reflect.Value, opts FieldOptions) (err error) {
 	if safeIsNil(rv) {
-		_, err := w.Write(bytesNull)
+		_, err = w.Write(bytesNull)
 		return err
 	}
 
 	// If the type implements json.Marshaler, just
 	// automatically respect that and skip to it.
 	if rv.Type().Implements(marshalerType) {
-		blob, err := rv.Interface().(json.Marshaler).MarshalJSON()
+		var blob []byte
```

Comment:
I don't understand the reason for this change.

Reply:
So I can do this on the err:

```golang
    if printLog {
        spew.Printf("(e) encodeReflectBinary(info: %v, rv: %#v (%v), opts: %v)\n",
            info, rv.Interface(), rv.Type(), opts)
        defer func() {
            fmt.Printf("(e) -> err: %v\n", err)
        }()
    }
```

Reply:
Aye aye, so for the named/naked returns.